### PR TITLE
Lets you attach signalers to desk bells

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Allows you to attach signalers to desk bells. With a bell in hand, use a signaler on it.
If a signaler attached to a desk bell is on receive mode and is activated by a radio pulse, it'll ring the bell.
If a desk bell with a signaler attached is rung, it'll act like you just hit transmit on the radio.

Have fun!

## Why It's Good For The Game
This gives a fun way to add a little button for triggering signals. Maybe you could set something like this up with toxins, or give someone a big red button to push?
Either way, I think it'll make for some fun interactions.

## Images of changes

https://github.com/ParadiseSS13/Paradise/assets/89928798/30c611bb-1b9e-4a72-b36b-632139136478

https://github.com/ParadiseSS13/Paradise/assets/89928798/6f761c8d-cdab-4a0b-a946-e07e6ee34686

## Testing
Tried ringing bells and signaling bells

## Changelog
:cl:
add: Desk bells can now have remote signalers attached to them.
add: If a desk bell is rung with a signaler attached, it'll send a signal on the signaler's frequency.
add: If a signaler attached to a desk bell is signaled, it'll ring the bell (if it's not broken).
/:cl:

### CODE REVIEW

**Comment by SynthTwo** [comment]
I watched this idea come into fruition. Awful.

**Comment by ATP-Engineer** [comment]
Does this also apply to wheelchair bells?
Last chance to look at me hector...

**Comment by lewcc** [comment]
> Does this also apply to wheelchair bells? Last chance to look at me hector...

Do we even have wheelchair bells on our codebase, or is that /tg/ only?

**Comment by ATP-Engineer** [comment]
I thought there was a PR got merged for it here.

File: code/modules/paperwork/desk_bell.dm [hunk 1]
```diff
@@ -79,12 +126,21 @@
...
+			return TRUE
+		to_chat(user, "<span class='notice'>You remove [attached_signaler].</span>")
+		user.put_in_hands(attached_signaler)
+		attached_signaler = null
 
 /// Check if the clapper breaks, and if it does, break it
 /obj/item/desk_bell/proc/check_clapper(mob/living/user)
 	if(prob(times_rang / 50) && ring_cooldown_length)
-		to_chat(user, "<span class='notice'>You hear [src]'s clapper fall off of its hinge. Nice job, you broke it.</span>")
+		audible_message("<span class='notice'>You hear [src]'s clapper fall off its hinge.")
```
Comment (Burzah) [inline code/modules/paperwork/desk_bell.dm:141]:
```suggestion
		audible_message("<span class='notice'>You hear [src]'s clapper fall off its hinge.</span>")
```

**Review by Burzah — CHANGES_REQUESTED** [review]

**Comment by DGamerL** [comment]
> I thought there was a PR got merged for it here.

Doesn't seem like it

**Comment by ATP-Engineer** [comment]
Guess not (but when someone adds it for wheelchairs, it needs to be signaler compatible)

Can it be attached to bicycle bells? thats the one I probably mistook it for.

File: code/modules/paperwork/desk_bell.dm [hunk 2]
```diff
@@ -15,19 +15,66 @@
...
+		if(cur_turf)
+			forceMove(attached_signaler, get_turf(src))
+		else
+			qdel(attached_signaler)
+		UnregisterSignal(attached_signaler, COMSIG_ASSEMBLY_PULSED)
+		attached_signaler = null
+	return ..()
+
+/obj/item/desk_bell/attackby(obj/item/I, mob/user, params)
+	// can only attach it while in hand
```
Comment (1080pCat) [inline code/modules/paperwork/desk_bell.dm:38]:
```suggestion
	// can only attach its on your person
```
pretty sure hands =/= inventory. solely atleast

File: code/modules/paperwork/desk_bell.dm [hunk 3]
```diff
@@ -15,19 +15,66 @@
...
+/obj/item/desk_bell/proc/try_ring(mob/user)
 	if(ring_cooldown > world.time || !anchored)
 		return TRUE
-	if(!ring_bell(user))
+	if(!ring_bell(user) && user)
 		to_chat(user, "<span class='notice'>[src] is silent. Some idiot broke it.</span>")
 	ring_cooldown = world.time + ring_cooldown_length
 	return TRUE
 
+
```
Comment (1080pCat) [inline code/modules/paperwork/desk_bell.dm:70]:
```suggestion

```

**Review by 1080pCat — COMMENTED** [review]
Does this work vise versa? being able to trigger the bell with a signaler? or just the bell triggering the signaler

File: code/modules/assembly/assembly.dm [hunk 4]
```diff
@@ -24,28 +24,17 @@
 	var/wires = ASSEMBLY_WIRE_RECEIVE | ASSEMBLY_WIRE_PULSE
```
Comment (Burzah) [inline code/modules/assembly/assembly.dm:24]: Can you change cooldown to `var/cooldown = 0` 
It probably got changed log ago by mistake.

**Comment by lewcc** [comment]
> Does this work vise versa? being able to trigger the bell with a signaler? or just the bell triggering the signaler

This works both ways, you can ring the bell with a signaler as well

**Comment by lewcc** [comment]
I will address this I swear

**Review by Burzah — APPROVED** [review]

File: code/modules/paperwork/desk_bell.dm [hunk 5]
```diff
@@ -15,19 +15,65 @@
...
+	var/obj/item/assembly/signaler/attached_signaler
 
-/obj/item/desk_bell/attack_hand(mob/living/user)
-	if(in_inventory && ishuman(user))
-		if(!user.get_active_hand())
-			user.put_in_hands(src)
-			return TRUE
+/obj/item/desk_bell/examine(mob/user)
+	. = ..()
+	if(!isnull(attached_signaler))
```
Comment (DGamerL) [inline code/modules/paperwork/desk_bell.dm:23]:
```suggestion
	if(attached_signaler)
```

File: code/modules/paperwork/desk_bell.dm [hunk 6]
```diff
@@ -15,19 +15,65 @@
...
+		. += "<span class='notice'>There seems to be an antenna sticking out of the base.</span>"
+
+/obj/item/desk_bell/Destroy()
+	if(!isnull(attached_signaler))
```
Comment (DGamerL) [inline code/modules/paperwork/desk_bell.dm:27]:
```suggestion
	if((attached_signaler)
```

File: code/modules/paperwork/desk_bell.dm [hunk 7]
```diff
@@ -15,19 +15,65 @@
...
+		attached_signaler = null
+	return ..()
+
+/obj/item/desk_bell/attackby(obj/item/I, mob/user, params)
+	// can only attach its on your person
+	if(istype(I, /obj/item/assembly/signaler))
+		if(!in_inventory)
+			to_chat(user, "<span class='warning'>[src] needs to be in your inventory if you want to attach [I] to it!</span>")
+			return
+		if(!isnull(attached_signaler))
```
Comment (DGamerL) [inline code/modules/paperwork/desk_bell.dm:43]:
```suggestion
		if((attached_signaler)
```

**Review by DGamerL — APPROVED** [review]
LGTM

**Pull request merged**